services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/paatos/PaatosService.kt

```diff
@@ -21,6 +22,11 @@ class PaatosService(
     fun findByHakemusId(hakemusId: Long): List<Paatos> =
         paatosRepository.findByHakemusId(hakemusId).map { it.toDomain() }
 
+    @Transactional
```

Comment:
Why `@Transactional`? Why not also in `saveKaivuilmoituksenPaatos`? Or maybe without it in both? Calling this function originates (besides the unit tests) from HakemusService.handleHakemusUpdate which is tagged as `@Transactional` - isn't that enough?

Reply:
I made this differently at first. I loaded the päätökset with the given hakemustunnus from the DB, then had a for-loop updating the status of each päätös. This needed `@Transactional` to persist the changes. The integration tests didn't work without it. The annotation is a remnant of that.

Now there's an explicit update statement - and only one statement - so the transaction is no longer required.

I think public methods should be marked as `@Transactional` if they need to be run in a transaction. That way there are no surprises if the caller is changed or the method is called from someplace else. There's very little downside. The default propagation is `REQUIRED`, which means any existing transaction will be used.

It's not completely clear to me if we should use `@Transactional` in all methods that call the database. Some people think it's never harmful, so it should be used for safety. Marking all database methods as either `@Transactional` or `@Transactional(readOnly=true)`

I'll use this approach here, so I'll mark all the public methods as `@Transactional`.